app/src/main/java/com/example/taskyy/ui/viewmodels/AgendaViewModel.kt

```diff
@@ -103,6 +104,15 @@ class AgendaViewModel @Inject constructor(
                 deleteReminder(event.agendaEventItem)
             }
             is AgendaEvent.EditExistingReminder -> {}
+            is AgendaEvent.StartWorkManager -> {
+                startWorkManager(event.context)
+            }
+        }
+    }
+
+    private fun startWorkManager(context: Context) {
```

Comment:
Also be very careful with using context references in the VM, especially the Activity context. If you store this in a field in the ViewModel (which you don't do here), this quickly causes a memory leak when the activity is destroyed due to a screen rotation for example.